### DISCUSSION TITLE
How to get an average value fom an alalog ADC input?

### DISCUSSION THREAD

**Discussion by sternegugger** [arendst/Tasmota#14517] [Q&A] [answered]
I would like to get an arithmetic mean value (average) from the ADC input over a time (e.g. 300 s).
I have a noisy analog input signal which should be smoothed.

Is there a configuration parameter to get an average value from ADC input over the measure time?

**Comment by sfromis** [comment, marked as answer]
No standard feature for that. Using rules, you could average over a *few* samples. There also exist algorithms for a moving average without storing a history of values. You could also, with ESP32, do calculations in Berry.

**Reply by pkkrusty** [reply]
Could also use influxdb integration and then do the averaging in Grafana
